pkg/cmd/pipelinerun/logs_test.go

```diff
@@ -2450,6 +2518,627 @@ func TestLog_pipelinerun_only_one_v1beta1(t *testing.T) {
+			},
+			Spec: v1beta1.PipelineRunSpec{
+				PipelineRef: &v1beta1.PipelineRef{
+					Name: pipelineName,
+				},
+			},
+			Status: v1beta1.PipelineRunStatus{
+				Status: duckv1beta1.Status{
+					Conditions: duckv1beta1.Conditions{
+						{
+							Status: corev1.ConditionTrue,
+							Reason: v1beta1.PipelineRunReasonRunning.String(),
+						},
+					},
+				},
+				PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+					TaskRuns: map[string]*v1beta1.PipelineRunTaskRunStatus{
+						tr1Name: {
+							PipelineTaskName: task1Name,
+							Status:           &trs[0].Status,
+						},
+						finallyTrName: {
+							PipelineTaskName: finallyName,
+							Status:           &trs[1].Status,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	pps := []*v1beta1.Pipeline{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pipelineName,
+				Namespace: ns,
+			},
+			Spec: v1beta1.PipelineSpec{
+				Tasks: []v1beta1.PipelineTask{
+					{
+						Name: task1Name,
+						TaskRef: &v1beta1.TaskRef{
+							Name: task1Name,
+						},
+					},
+				},
+				Finally: []v1beta1.PipelineTask{
+					{
+						Name: finallyName,
+						TaskRef: &v1beta1.TaskRef{
+							Name: finallyName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	p := []*corev1.Pod{
+		tb.Pod(tr1Pod,
+			tb.PodNamespace(ns),
+			tb.PodLabel("tekton.dev/task", pipelineName),
+			tb.PodSpec(
+				tb.PodContainer(tr1Step1Name, tr1Step1Name+":latest"),
+			),
+			cb.PodStatus(
+				cb.PodPhase(corev1.PodSucceeded),
+			),
+		),
+		tb.Pod(finallyTrPod,
+			tb.PodNamespace(ns),
+			tb.PodLabel("tekton.dev/task", pipelineName),
+			tb.PodSpec(
+				tb.PodContainer(finallyTrStep1Name, finallyTrStep1Name+":latest"),
+			),
+			cb.PodStatus(
+				cb.PodPhase(corev1.PodSucceeded),
+			),
+		),
+	}
+
+	fakeLogStream := fake.Logs(
+		fake.Task(tr1Pod,
+			fake.Step(tr1Step1Name, "wrote a file1"),
+		),
+		fake.Task(finallyTrPod,
+			fake.Step(finallyTrStep1Name, "Finally"),
+		),
+	)
+
+	cs, _ := test.SeedV1beta1TestData(t, pipelinev1beta1test.Data{PipelineRuns: prs, Pipelines: pps, TaskRuns: trs, Pods: p, Namespaces: nsList})
+	cs.Pipeline.Resources = cb.APIResourceList(versionB1, []string{"task", "taskrun", "pipeline", "pipelinerun"})
+
+	tdc := testDynamic.Options{}
+	dc, err := tdc.Client(
+		cb.UnstructuredV1beta1TR(trs[0], versionB1),
+		cb.UnstructuredV1beta1TR(trs[1], versionB1),
+		cb.UnstructuredV1beta1PR(prs[0], versionB1),
+		cb.UnstructuredV1beta1P(pps[0], versionB1),
+	)
+
+	if err != nil {
+		t.Errorf("unable to create dynamic client: %v", err)
+	}
+
+	prlo := logOptsv1beta1(prName, ns, cs, dc, fake.Streamer(fakeLogStream), false, false)
+	output, _ := fetchLogs(prlo)
+
+	expectedLogs := []string{
+		"[output-task : writefile-step] wrote a file1\n",
+		"[finally-task : finally-step] Finally\n",
+	}
+	expected := strings.Join(expectedLogs, "\n") + "\n"
+	test.AssertOutput(t, expected, output)
+}
+
+/*func TestPipelinerunLog_with_follow_mode_finally_task_v1beta1(t *testing.T) {
```

Comment:
We should removed this code if commented :wink:

Reply:
And write an e2e tests instead for it :stuck_out_tongue_closed_eyes:

Reply:
Yeh :stuck_out_tongue_closed_eyes:
Changed this to e2e